### PR TITLE
Week3 Step 3 Finished

### PR DESCRIPTION
**Pull request by Xin128**
1) Copied the original pom.xml and edit the original appengine-web.xml for deploy
2) Resolve the name of "Greetings"
3) Implementation of DataServelet.java to achieve transform to Json
4) Passed the Java Linter

### CODE REVIEW

File: portfolio/pom.xml
Comment (cisxingg) [inline portfolio/pom.xml]: Extra line?

File: portfolio/pom.xml
Comment (cisxingg) [inline portfolio/pom.xml]: Still a todo? Can remove if it is not the case anymore.

File: portfolio/pom.xml
Comment (cisxingg) [inline portfolio/pom.xml]: Extra lines?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (cisxingg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: "returns" seems fine?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (cisxingg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Question: Is this needed to call convertArrayToJsonUsingGson()?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (cisxingg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nit: Sentences that start with "Note" can start in a new line.

File: portfolio/src/main/webapp/script.js
Comment (cisxingg) [inline portfolio/src/main/webapp/script.js]: Not a great function name :D

File: portfolio/pom.xml
Comment (warfa-google) [inline portfolio/pom.xml]: You can remove this TODO

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (warfa-google) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: returns

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (warfa-google) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Whats a ServerStats instance? Doesn't this convert an arraylist?

File: portfolio/src/main/webapp/script.js
Comment (warfa-google) [inline portfolio/src/main/webapp/script.js]: Is this code used? Should we remove it

File: portfolio/src/main/webapp/script.js
Comment (cisxingg) [inline portfolio/src/main/webapp/script.js]: Do you mean to call this function somewhere?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java
Comment (Xin128) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: They did wrap it in the sample code, which I also don't think is quite needed. I can directly convert this to a line of code

File: portfolio/src/main/webapp/script.js
Comment (Xin128) [reply portfolio/src/main/webapp/script.js]: This is just from step 2 practice. I just removed it.

File: portfolio/src/main/webapp/script.js
Comment (Xin128) [reply portfolio/src/main/webapp/script.js]: This is just from step 2 practice. I just removed it.

**Review by cisxingg — APPROVED** [review]
LGTM, thanks for resolving the comments! Please wait for Warfa's approval before merging.

**Review by warfa-google — APPROVED** [review]

**Pull request merged**
